platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/collections/DefaultConfigurableFileCollection.java

```diff
@@ -675,6 +675,9 @@ public List<Object> replace(FileCollectionInternal original, Supplier<FileCollec
             ImmutableList.Builder<Object> builder = ImmutableList.builderWithExpectedSize(items.size());
             boolean hasChanges = false;
             for (Object candidate : items) {
+                if (candidate == null) {
+                    continue;
```

Comment:
💭  This is a direct source of silent misconfiguration. I know we [decided](https://github.com/gradle/gradle/issues/32553#issuecomment-2687837464) to fix the NPE first and figure out nulls later, but I'd much prefer we state this in a comment that this is more of a workaround, rather than a long-term decision.

It'd be best to have a dedicated issue already.